scriptmodules/supplementary/emulationstation/configscripts/mupen64plus.sh

```diff
@@ -0,0 +1,260 @@
+    for key in "${keys[@]}"; do
+        # read key value. Axis takes two key/axis values.
+        iniGet "$key"
+        case "$input_type" in
+            axis) 
+                # key "X/Y Axis" needs different button naming
+                if [[ $key == *"Axis"* ]]; then
+                    # if there is already a "-" axis add "+" axis value
+                    if   [[ $ini_value == *"("* ]]; then
+                        value="${ini_value}${input_id}+)"
+                    # if there is already a "+" axis add "-" axis value
+                    elif [[ $ini_value == *")"* ]]; then
+                        value="axis(${input_id}-,${ini_value}"
+                    # if there is no ini_value add "+" axis value
+                    elif [[ "$input_value" == "1" ]]; then
+                        value="${input_id}+)"
+                    else 
+                        value="axis(${input_id}-,"
+                    fi
+                elif [[ "$input_value" == "1" ]]; then
+                    value="axis(${input_id}+) ${ini_value}"
+                else
+                    value="axis(${input_id}-) ${ini_value}"
+                fi
+                ;;
+            hat)
+                if [[ $key == *"Axis"* ]]; then
+                    if   [[ $ini_value == *"("* ]]; then
+                        value="${ini_value}${input_id} ${dir})"
+                    elif [[ $ini_value == *")"* ]]; then
+                        value="axis(${input_id} ${dir},${ini_value}"
+                    elif [[ "$dir" == "Up" ]] || [[ "$dir" == "Left" ]]; then
+                        value="button(${input_id} ${dir},"
+                    elif [[ "$dir" == "Right" ]] || [[ "$dir" == "Down" ]]; then
+                        value="${input_id} ${dir})"
+                    fi
+                else
+                    value="hat(${input_id} ${dir}) ${ini_value}"
+                fi
+                ;;
+            *)
+                if [[ $key == *"Axis"* ]]; then
+                    if   [[ $ini_value == *"("* ]]; then
+                        value="${ini_value}${input_id})"
+                    elif [[ $ini_value == *")"* ]]; then
+                        value="button(${input_id},${ini_value}"
+                    elif [[ "$dir" == "Up" ]] || [[ "$dir" == "Left" ]]; then
+                        value="button(${input_id},"
+                    elif [[ "$dir" == "Right" ]] || [[ "$dir" == "Down" ]]; then
+                        value="${input_id})"
+                    fi
+                else
+                    value="button(${input_id}) ${ini_value}"
+                fi
+                ;;
+        esac
+
+        iniSet "$key" "$value"
+    done
+}
+
+function onend_mupen64plus_joystick() {
+    local device_type=$1
+    local device_name=$2
+    
+    local bind
+    local axis_neg
+    local axis_pos
+    for axis in "X Axis" "Y Axis"; do
+        if [[ $axis == *"X"* ]]; then
+            axis_neg="DPad L"
```

Comment:
But you are right. 
Strange. Why does it run anyway? I have not seen a error message and remapping of DPAD to AXIS works?